src/pages/workspaces/workspace/Dashboard/WorkspaceTags.ts

```diff
@@ -0,0 +1,119 @@
+import { InfoBox, Link, Spinner } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import { CSSProperties, ReactNode, useState } from 'react';
+import { div, h, i, span } from 'react-hyperscript-helpers';
+import { icon } from 'src/components/icons';
+import { WorkspaceTagSelect } from 'src/components/workspace-utils';
+import { Ajax } from 'src/libs/ajax';
+import { getEnabledBrand } from 'src/libs/brand-utils';
+import colors from 'src/libs/colors';
+import { withErrorReporting } from 'src/libs/error';
+import * as Style from 'src/libs/style';
+import { withBusyState } from 'src/libs/utils';
+import { InitializedWorkspaceWrapper as Workspace } from 'src/pages/workspaces/hooks/useWorkspace';
+import { RightBoxSection } from 'src/pages/workspaces/workspace/Dashboard/RightBoxSection';
+
+const styles: Record<string, CSSProperties> = {
+  authDomain: {
+    padding: '0.5rem 0.25rem',
+    marginBottom: '0.25rem',
+    backgroundColor: colors.dark(0.15),
+    ...Style.noWrapEllipsis,
+  },
+  tag: {
+    padding: '0.25rem',
+    margin: '0.15rem',
+    backgroundColor: colors.dark(0.15),
+    borderRadius: 10,
+    overflow: 'hidden',
+    wordWrap: 'break-word',
+  },
+};
+
+interface WorkspaceTagsProps {
+  workspace: Workspace;
+  canEdit: boolean;
+}
+
+export const WorkspaceTags = (props: WorkspaceTagsProps): ReactNode => {
+  const { workspace, canEdit } = props;
+  const { namespace, name } = workspace.workspace;
+
+  const initialTags: string[] = getWorkspaceTags(workspace);
+
+  const [busy, setBusy] = useState<boolean>(false);
+  const [tagsList, setTagsList] = useState<string[]>(initialTags);
+
+  const persistenceId = `workspaces/${namespace}/${name}/dashboard/tagsPanelOpen`;
+
+  const addTag = _.flow(
+    withErrorReporting('Error adding tag'),
+    withBusyState(setBusy)
+  )(async (tag) => {
+    setTagsList(await Ajax().Workspaces.workspace(namespace, name).addTag(tag));
+  });
+
+  const deleteTag = _.flow(
+    withErrorReporting('Error removing tag'),
+    withBusyState(setBusy)
+  )(async (tag) => {
+    setTagsList(await Ajax().Workspaces.workspace(namespace, name).deleteTag(tag));
+  });
+
+  const brand = getEnabledBrand();
+
+  return h(
+    RightBoxSection,
+    {
+      title: 'Tags',
+      info: span({}, [(busy || !tagsList) && h(Spinner, { size: 1, style: { marginLeft: '0.5rem' } })]),
```

Comment:
This component is still managing its own collapse, because it uses component state for the `info` component.